tasks/get_current_archive_list/get_current_archive_list.py

```diff
@@ -486,13 +471,34 @@ def translate_s3_import_to_partitioned_data_sql(report_table_name: str) -> TextC
     """
     return text(
         f"""
-        INSERT INTO orca.{report_table_name} (job_id, orca_archive_location, key_path, etag, last_update, size_in_bytes, storage_class)
-            SELECT :job_id, orca_archive_location, key_path, etag, last_update, size_in_bytes, storage_class
+        INSERT INTO orca.{report_table_name} (job_id, orca_archive_location, key_path, etag, last_update, size_in_bytes, storage_class, delete_flag)
+            SELECT :job_id, orca_archive_location, key_path, etag, last_update, size_in_bytes, storage_class, delete_flag
             FROM s3_import
         """
     )
 
 
+def get_s3_credentials_from_secrets_manager() -> tuple:
+    # todo: Move everything from here to get_configuration to shared lib. See shared_db for code origin.
+    prefix = os.getenv("PREFIX", None)
+    secretsmanager = boto3.client("secretsmanager", region_name=os.getenv("AWS_REGION", None))
+    s3_credentials = json.loads(
+        secretsmanager.get_secret_value(SecretId=f"{prefix}-orca-{SECRETSMANAGER_S3_ACCESS_CREDENTIALS_KEY}")[
+            "SecretString"
+        ]
+    )
+    s3_access_key = s3_credentials.get(S3_ACCESS_CREDENTIALS_ACCESS_KEY_KEY, None)
+    if s3_access_key is None or len(s3_access_key) == 0:
+        LOGGER.error(f"{S3_ACCESS_CREDENTIALS_ACCESS_KEY_KEY} secret is not set.")
+        raise KeyError(f"{S3_ACCESS_CREDENTIALS_ACCESS_KEY_KEY} secret is not set.")
```

Comment:
I wonder if this should be a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) instead of a key error.

Reply:
Swapped